parcels/tools/converters.py

```diff
@@ -57,10 +57,13 @@ def __init__(self, time_origin: float | np.datetime64 | np.timedelta64 | cftime.
         self.time_origin = time_origin
         self.calendar: str | None = None
         if isinstance(time_origin, np.datetime64):
+            self.time_origin = time_origin
             self.calendar = "np_datetime64"
         elif isinstance(time_origin, np.timedelta64):
+            self.time_origin = time_origin.astype("timedelta64[s]")
```

Comment:
Is it a concern that this may lose subsecond resolution?

```
>>> np.timedelta64(110, "ns").astype('timedelta64[s]')
np.timedelta64(0,'s')
```

Reply:
Now using nanoseconds for timedelta64; better like this?

Reply:
The issue stems with the actual comparison of datetime objects in `.reltime()`. I'm not entirely sure what datetime objects are coming into (or expected to come into) that function, so something more sophisticated may be needed.

Reply:
OK is fine; then let's discuss in person. Or feel free to propose something else, if you want